### PR TITLE
[Trivial][Tests] Do not fail test when warnings are written to stderr

### PR DESCRIPTION
**Pull request by random-zebra**
Fixes https://github.com/PIVX-Project/PIVX/issues/960

Rationale: the test result should be set only through `assert`s in the test and not rely on what's printed on the console. Furthermore, as shown in the issue, stderr could contain simple warnings that don't affect the test's outcome.
If stderr is not empty though, print its content.

Also re-sort tests by duration.

### CODE REVIEW

**Comment by CaveSpectre11** [comment]
utACK

**Comment by akshaynexus** [comment]
ACK ,tested on the pr which had this issue,travis shows checks successful.

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -306,7 +310,11 @@ def run_tests(test_list, src_dir, build_dir, exeext, tmpdir, jobs=1, enable_cove
         test_results.append(test_result)
         done_str = "{}/{} - {}{}{}".format(i + 1, test_count, BOLD[1], test_result.name, BOLD[0])
         if test_result.status == "Passed":
-            logging.debug("%s passed, Duration: %s s" % (done_str, test_result.time))
+            if stderr == "":
+                logging.debug("%s passed, Duration: %s s\n" % (done_str, test_result.time))
```
Comment (Fuzzbawls) [inline test/functional/test_runner.py]: nit: remove the newline `\n` here to reduce needless spacing

File: test/functional/test_runner.py [hunk 2]
```diff
@@ -306,7 +310,11 @@ def run_tests(test_list, src_dir, build_dir, exeext, tmpdir, jobs=1, enable_cove
...
+            else:
+                logging.debug("%s passed (with warnings), Duration: %s s\n" % (done_str, test_result.time))
+                print(BOLD[1] + 'stderr:\n' + BOLD[0] + stderr + '\n')
```
Comment (Fuzzbawls) [inline test/functional/test_runner.py]: nit: remove the last newline `\n` here to reduce needless spacing

**Pull request merged**